milli/src/search/distinct/facet_distinct.rs

```diff
@@ -0,0 +1,203 @@
+use std::mem::size_of;
+
+use roaring::RoaringBitmap;
+
+use crate::heed_codec::facet::*;
+use crate::{facet::FacetType, DocumentId, FieldId, Index};
+use super::{Distinct, DocIter};
+
+/// A distinct implementer that is backed by facets. On each iteration, the facet values for the
+/// distinct attribute of the first document are retrieved. The document ids for these facet values
+/// are then retrieved and taken out of the the candidate and added to the excluded set. We take
+/// care to keep the document we are currently on, and remove it from the excluded list. The next
+/// iterations will never contain any occurence of a document with the same distinct value as a
+/// document from previous iterations.
```

Comment:
```suggestion
/// A distinct implementer that is backed by facets.
/// 
/// On each iteration, the facet values for the distinct attribute of the first document
/// are retrieved. The document ids for these facet values are then retrieved and taken
/// out of the candidate and added to the excluded set. We take care to keep
/// the document we are currently on, and remove it from the excluded list. The next
/// iterations will never contain any occurrence of a document with the same distinct
/// value as a document from previous iterations.
```

I just isolated the first sentence to make it the only sentence visible by default in the Rust doc.